.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,127 @@
+          - docker-image/v1.16/arm64/debian-logentries
+          - docker-image/v1.16/arm64/debian-loggly
+          - docker-image/v1.16/arm64/debian-logzio
+          - docker-image/v1.16/arm64/debian-opensearch
+          - docker-image/v1.16/arm64/debian-papertrail
+          - docker-image/v1.16/arm64/debian-s3
+          - docker-image/v1.16/arm64/debian-syslog
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          platforms: linux/amd64,linux/arm64
+      - uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      - name: Setup tags
+        run: |
+          set -x
+          context=${{ matrix.context }}
+          component=${context##*/}
+          for target in $(make echo-all-images); do
+            case $target in
+              *$component-amd64*)
+                tags=$(echo $target | cut -d':' -f2-)
+                tag1=$(echo $tags | cut -d',' -f1)
+                tag2=$(echo $tags | cut -d',' -f2)
+                echo "AMD64TAGS=${{ env.REPOSITORY }}:${tag1},${{ env.REPOSITORY }}:${tag2}" >> ${GITHUB_ENV}
+                echo "MULTIARCH_TAG=${tag1/amd64-/}" >> ${GITHUB_ENV}
+                echo "MULTIARCH_SHORT_TAG=${tag2/amd64-/}" >> ${GITHUB_ENV}
+                echo "AMD64TAG=${tag1}" >> ${GITHUB_ENV}
+                echo "SHORT_AMD64TAG=${tag2}" >> ${GITHUB_ENV}
+                ;;
+              *$component-arm64*)
+                tags=$(echo $target | cut -d':' -f2-)
+                tag1=$(echo $tags | cut -d',' -f1)
+                tag2=$(echo $tags | cut -d',' -f2)
+                echo "ARM64TAGS=${{ env.REPOSITORY }}:${tag1},${{ env.REPOSITORY }}:${tag2}" >> ${GITHUB_ENV}
+                echo "ARM64TAG=${tag1}" >> ${GITHUB_ENV}
+                echo "SHORT_ARM64TAG=${tag2}" >> ${GITHUB_ENV}
+                ;;
+            esac
+          done
+          amdcontext=${context//arm64\//}
+          echo "AMDCONTEXT=${amdcontext}" >> ${GITHUB_ENV}
+      - name: Check buildx version
+        run: docker buildx version
+      - name: Build and push for amd64
+        uses: docker/build-push-action@v6
+        with:
+          context: ${{ env.AMDCONTEXT }}
+          provenance: false
+          push: true
+          platforms: linux/amd64
+          tags: ${{ env.AMD64TAGS }}
+          # dare to use old mediatype (application/vnd.docker.distribution.manifest.v2+json)
+          outputs: oci-mediatypes=false
+      - name: Build and push for arm64
+        uses: docker/build-push-action@v6
+        with:
+          context: ${{ matrix.context }}
+          provenance: false
+          push: true
+          platforms: linux/arm64
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```

Comment:
I'm not sure about this cache mechanism, but don't we also need it for `amd64`?

Reply:
From the reference, we need to specify the platforms like as:  `platforms: linux/amd64,linux/arm64`

ref: https://docs.docker.com/build/ci/github-actions/cache/#cache-mounts

Reply:
Thanks, I'm not sure how to integrate with it yet, so keep it as a further improvements.